### PR TITLE
[ENH] MNEAnalyze saves last folder where data were loaded from

### PR DESCRIPTION
**Pull request by juangpc**
A little bit of refactoring in the way in which the QSettings was used by mainwindow to store the dark theme and a few other info.
Also, a few warnings addressed, not solved. but at least, not showing without any chance of actually breaking the code. 
And finally, added a much needed feature of storing the directory from where the last fif file was loaded.

### CODE REVIEW

File: applications/mne_analyze/libs/anShared/Model/bemdatamodel.cpp [hunk 1]
```diff
@@ -113,8 +113,9 @@ void BemDataModel::initBemData(QIODevice& qIODevice)
 
 //=============================================================================================================
 
-QVariant BemDataModel::data(const QModelIndex &index,
-                            int role) const
+QVariant BemDataModel::data(const QModelIndex &,//index,
+                            int )//role)
+                            const
```
Comment (LorenzE) [inline applications/mne_analyze/libs/anShared/Model/bemdatamodel.cpp:118]: Is there a specific reason why the variables were removed?

File: applications/mne_analyze/libs/anShared/Model/fiffrawviewmodel.h [hunk 2]
```diff
@@ -911,7 +911,7 @@ class AnnotationData {
 private:
 
     int m_iSample;
-    int m_iEventType;
+    //int m_iEventType;
```
Comment (LorenzE) [inline applications/mne_analyze/libs/anShared/Model/fiffrawviewmodel.h:914]: Is this member used anywhere? If not, we should consider removing it completley.

File: applications/mne_analyze/mne_analyze/mainwindow.h [hunk 3]
```diff
@@ -191,6 +191,14 @@ class MainWindow : public QMainWindow
...
     void createLogDockWindow();
 
+    //=========================================================================================================
+    /**
+     * Setter method for CurrentStyle member variable.
+     */
+    void setCurrentStyle(const QString& style);
+
+
+
```
Comment (LorenzE) [inline applications/mne_analyze/mne_analyze/mainwindow.h:201]: remove two of the three empty lines.

File: applications/mne_analyze/plugins/dataloader/dataloader.h [hunk 4]
```diff
@@ -176,6 +178,25 @@ class DATALOADERSHARED_EXPORT DataLoader : public ANSHAREDLIB::AbstractPlugin
...
+     */
+    void loadSettings();
+
+    //=========================================================================================================
+    /**
+     * Save the settings in the register/file/mechanism for lon-term storage.
+     */
+    void saveSettings();
+
+
```
Comment (LorenzE) [inline applications/mne_analyze/plugins/dataloader/dataloader.h:193]: remove one empty line

File: applications/mne_analyze/plugins/datamanager/datamanager.cpp [hunk 5]
```diff
@@ -144,11 +144,12 @@ QWidget *DataManager::getView()
...
-    default:
-        qWarning() << "[DataManager::handleEvent] received an Event that is not handled by switch-cases";
-        break;
-    }
+    Q_UNUSED(e)
+//    switch (e->getType()) {
+//    default:
+//        qWarning() << "[DataManager::handleEvent] received an Event that is not handled by switch-cases";
+//        break;
+//    }
```
Comment (LorenzE) [inline applications/mne_analyze/plugins/datamanager/datamanager.cpp:152]: Why was this commented out?

File: libraries/disp/plots/helpers/colormap.cpp [hunk 6]
```diff
@@ -316,7 +316,7 @@ int ColorMap::coolG(double x)
 
 //=============================================================================================================
 
-int ColorMap::coolB(double x)
+int ColorMap::coolB(double /*x*/)
```
Comment (LorenzE) [inline libraries/disp/plots/helpers/colormap.cpp:319]: See first comment

File: libraries/disp/viewers/channelselectionview.cpp [hunk 7]
```diff
@@ -410,7 +410,7 @@ void ChannelSelectionView::updateDataView()
                 i.remove();
             }
         }
-        SelectionSceneItem* test = static_cast<SelectionSceneItem*>(visibleItemList.first());
+        //SelectionSceneItem* test = static_cast<SelectionSceneItem*>(visibleItemList.first());
```
Comment (LorenzE) [inline libraries/disp/viewers/channelselectionview.cpp:413]: If this variable is no longer needed, we should consider removing it.

**Review by LorenzE — COMMENTED** [review]
some minor comments/questions

File: applications/mne_analyze/libs/anShared/Model/bemdatamodel.cpp (hunk 1 above)
Comment (juangpc) [reply applications/mne_analyze/libs/anShared/Model/bemdatamodel.cpp:118]: This and analogous edits are really in the interest of making the warnings at least not disturb and distract while we are developing. There are aprox. 150 now. Mainly in areas of the codebase where we're not working at the moment. So if we are able to achieve zero warnings, then we would really take matters more seriously whenever there are in the areas of the code where we are working. That is my -broken?- train of though. 

Regarding this precise edit. Q_UNUSED(VAR) generates another warning which only stops when you remove the var name in the method definition.

File: applications/mne_analyze/libs/anShared/Model/fiffrawviewmodel.h (hunk 2 above)
Comment (juangpc) [reply applications/mne_analyze/libs/anShared/Model/fiffrawviewmodel.h:914]: I don't want to undo anybody's work. just want the <<noise>> to go away 😅

File: applications/mne_analyze/plugins/datamanager/datamanager.cpp (hunk 5 above)
Comment (juangpc) [reply applications/mne_analyze/plugins/datamanager/datamanager.cpp:152]: After a long conversaition out in the RealWrold, it was simply a mistake on my side to comment it. I'll undo the changes.

File: applications/mne_analyze/mne_analyze/mainwindow.h (hunk 3 above)
Comment (juangpc) [reply applications/mne_analyze/mne_analyze/mainwindow.h:201]: done

File: applications/mne_analyze/plugins/dataloader/dataloader.h (hunk 4 above)
Comment (juangpc) [reply applications/mne_analyze/plugins/dataloader/dataloader.h:193]: done

File: applications/mne_analyze/plugins/datamanager/datamanager.cpp (hunk 5 above)
Comment (juangpc) [reply applications/mne_analyze/plugins/datamanager/datamanager.cpp:152]: undone

File: libraries/disp/plots/helpers/colormap.cpp (hunk 6 above)
Comment (juangpc) [reply libraries/disp/plots/helpers/colormap.cpp:319]: undone

File: libraries/disp/viewers/channelselectionview.cpp (hunk 7 above)
Comment (juangpc) [reply libraries/disp/viewers/channelselectionview.cpp:413]: undone

Comment (juangpc) [reply libraries/disp/viewers/channelselectionview.cpp:413]: I think that there is a "better" way.

**Comment by juangpc** [comment]
Hey, sorry for the back and forth. 
The pr is focused on the addition of the previously used path in mne-analyze. (plus some very minor changes to delete warnings). 
Much less intrusive than before.

File: applications/mne_analyze/plugins/dataloader/dataloader.cpp [hunk 8]
```diff
@@ -259,21 +262,55 @@ void DataLoader::onLoadFilePressed()
...
+{
+    m_sLastDir = lastDir;
+    saveSettings();
+}
+
+//=============================================================================================================
+
+void DataLoader::saveSettings()
+{
+    QSettings settings("MNECPP");
```
Comment (LorenzE) [inline applications/mne_analyze/plugins/dataloader/dataloader.cpp:293]: Please use `MNEANALYZE/DATALOADER` instead of `MNECPP`

**Review by LorenzE — CHANGES_REQUESTED** [review]

File: applications/mne_analyze/plugins/dataloader/dataloader.cpp (hunk 8 above)
Comment (juangpc) [reply applications/mne_analyze/plugins/dataloader/dataloader.cpp:293]: This "MNECPP" is the top key value under which every setting/data is stored in the win register(win)/config files(unix). It is used in mainwindow, several other plugins, etc...

**Review by LorenzE — APPROVED** [review]
Looks good!

**Review by gabrielbmotta — APPROVED** [review]

**Pull request merged**